web/src/components/FeedbackModal/FeedbackModal.test.tsx

```diff
@@ -0,0 +1,294 @@
+      mockApi.postFeedback.mockRejectedValue(500);
+
+      renderFeedbackModal();
+
+      fireEvent.click(screen.getByText(Config.feedbackModal.betaModalFeatureRequestButtonText));
+      fireEvent.change(screen.getByLabelText("Feature request"), { target: { value: "random text" } });
+      fireEvent.click(screen.getByText(Config.feedbackModal.feedbackModalSubmitButtonText));
+
+      await waitFor(() => {
+        expect(screen.getByTestId("dialog-alert")).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("report an issue", () => {
+    it("does not show report issue button when userData is undefined", () => {
+      useUndefinedUserData();
+      renderFeedbackModal();
+      expect(screen.queryByText(Config.feedbackModal.betaModalBugReportButtonText)).not.toBeInTheDocument();
+    });
+
+    it("renders report issue modal and updates input elements", async () => {
+      renderFeedbackModal();
+
+      fireEvent.click(screen.getByText(Config.feedbackModal.betaModalBugReportButtonText));
+      fireEvent.change(screen.getByLabelText("Issue summary"), { target: { value: "random summary text" } });
+      fireEvent.change(screen.getByLabelText("Issue details"), { target: { value: "random details text" } });
+
+      expect(screen.getByText(Config.feedbackModal.reportBugModalHeadingText)).toBeInTheDocument();
+      expect((screen.getByLabelText("Issue summary") as HTMLInputElement).value).toBe("random summary text");
+      expect((screen.getByLabelText("Issue details") as HTMLInputElement).value).toBe("random details text");
+    });
+
+    it("does not submit reported issue when issue summary input is empty and shows inline validation error", () => {
+      renderFeedbackModal();
+
+      fireEvent.click(screen.getByText(Config.feedbackModal.betaModalBugReportButtonText));
+      fireEvent.change(screen.getByLabelText("Issue summary"), { target: { value: "random summary text" } });
+      fireEvent.click(screen.getByText(Config.feedbackModal.feedbackModalSubmitButtonText));
+
+      expect(screen.getByText(Config.feedbackModal.feedbackModalInlineErrorText)).toBeInTheDocument();
+      expect(mockApi.postIssue).not.toHaveBeenCalled();
+    });
+
+    it("does not submit reported issue when issue detail input is empty and shows inline validation error", () => {
+      renderFeedbackModal();
+
+      fireEvent.click(screen.getByText(Config.feedbackModal.betaModalBugReportButtonText));
+      fireEvent.change(screen.getByLabelText("Issue details"), { target: { value: "random details text" } });
+      fireEvent.click(screen.getByText(Config.feedbackModal.feedbackModalSubmitButtonText));
+
+      expect(screen.getByText(Config.feedbackModal.feedbackModalInlineErrorText)).toBeInTheDocument();
+      expect(mockApi.postIssue).not.toHaveBeenCalled();
+    });
+
+    it("opens issues dialog when isBug prop is used and then returns to feedback selection dialog after issue submission", async () => {
+      renderFeedbackModal(true);
```

Comment:
for readability, I'd recommend using an object for the parameters here. `renderFeedbackModal({isBugBar: true})` is much easier to understand what the param is doing than just the naked boolean here